ui/src/infra-model/view/infra-model-view.tsx

```diff
@@ -112,7 +119,7 @@ export const InfraModelView: React.FC<InfraModelViewProps> = (props: InfraModelV
                         <InfraModelForm
                             changeTimes={props.changeTimes}
                             validationIssues={props.validationIssues}
-                            upLoading={props.isLoading}
+                            upLoading={props.isSaving}
```

Comment:
Tätä käytetään käytännössä siihen, että näytetään inframallinäkymän yläkulmassa teksti "Tiedostoa ladataan tietokantaan ...". `isLoading` laitetaan myös päälle aina kun lähetetään validaatiokutsu, ja tällöin ei tallenneta mitään kantaan -> erotettu tallentamisen tila omakseen

Reply:
Tilakombinaatio jossa isLoading on poissa päältä (false) ja isSaving on päällä (true) ei taida kuitenkaan olla mahdollinen? (Viittaisi ennemmin yhden tilatyypin käyttöön verrattuna kahteen booleaniin [eli esim. not_loading, validating, saving], johon voisi myös lisätä api_error-vaihtoehdot halutessaan)

Reply:
Periaatteessa ne tuskin ovat koskaan yhtä aikaa true, mutta käytännössä ne kuvaavat ihan eri asioita ja pitäisin ne siten itse erillään. Tosin tuon isLoading:in voisi kyllä nimetä paremmin isParsing/isValidating:iksi tms.

Reply:
🤔 Aa aivan, aivan, tuo isLoading ei siis ollutkaan kovin kuvaava. Oletettavasti käyttäjän on siis mahdollista yrittää tallentaa tiedosto ennen validaatiokutsun palautumista? Tuntuu hieman oudolta toimintaperiaatteelta, joka ehkä pitäisi blokkailla muutenkin (=> jolloin yhtenäistys samaan tilaan kuulostaisi omaan korvaan oikealta)

Reply:
Ei sen pitäisi olla nytkään mahdollista UI:n puolesta. Itse tosin silti jotenkin karsastan tuollaisia "tässä on koko UI:n yleistila redusoituna yhteen arvojoukkoon" -muuttujia. Mun kokemuksella ne menee todella usein siihen että sitten siellä on `"TÄMÄ_MUTTA_EI_TOINEN" | "EI_TÄMÄ_MUTTA_TOINEN"` -tyyppisiä tiloja, jolloin oikeampi ratkaisu olisi ollut jo lähtökohtaisesti erottaa ne vain omiksi inferroiduiksi jutuikseen

Reply:
Noo, nimeäminen on tietysti hankalaa ja tuollainen NOT_-prefiksaus on se helppo ja nopea veto. 

Itse taas karsastan liian laajan tila-avaruuden tukemista, joista ainoastaan osa kombinaatioista on realistisesti mahdollisia -> pitää tulkita kombinaatioiden erikoistapauksia joita ei edes ikinä käytetä